Sources/Networking/Responses/CustomerInfoResponse.swift

```diff
@@ -0,0 +1,190 @@
+
+    }
+
+}
+
+// MARK: -
+
+extension CustomerInfoResponse.Subscriber: Codable, Hashable {}
+extension CustomerInfoResponse.Entitlement: Codable, Hashable {}
+extension CustomerInfoResponse.Subscription: Codable, Hashable {}
+
+extension CustomerInfoResponse.Transaction: Codable, Hashable {
+
+    private enum CodingKeys: String, CodingKey {
+
+        case purchaseDate
+        case originalPurchaseDate
+        case transactionIdentifier = "id"
+        case store
+        case isSandbox
+
+    }
+
+}
+
+extension CustomerInfoResponse: Codable {}
+
+// Equality + hash ignore request date.
+extension CustomerInfoResponse: Equatable, Hashable {
+
+    static func == (lhs: CustomerInfoResponse, rhs: CustomerInfoResponse) -> Bool {
+        return lhs.subscriber == rhs.subscriber
+    }
+
+    func hash(into hasher: inout Hasher) {
+        self.subscriber.hash(into: &hasher)
+    }
+
+}
+
+extension CustomerInfoResponse.Transaction {
+
+    init(
+        purchaseDate: Date?,
+        originalPurchaseDate: Date?,
+        transactionIdentifier: String?,
+        store: Store,
+        isSandbox: Bool
+    ) {
+        self.purchaseDate = purchaseDate
+        self.originalPurchaseDate = originalPurchaseDate
+        self.transactionIdentifier = transactionIdentifier
+        self.store = store
+        self.isSandbox = isSandbox
+    }
+
+    var asSubscription: CustomerInfoResponse.Subscription {
+        return .init(purchaseDate: self.purchaseDate,
+                     originalPurchaseDate: self.originalPurchaseDate,
+                     store: self.store,
+                     isSandbox: self.isSandbox)
+    }
+
+}
+
+extension CustomerInfoResponse.Subscription {
+
+    init(
+        periodType: PeriodType = .defaultValue,
+        purchaseDate: Date? = nil,
+        originalPurchaseDate: Date? = nil,
+        expiresDate: Date? = nil,
+        store: Store = .defaultValue,
+        isSandbox: Bool,
+        unsubscribeDetectedAt: Date? = nil,
+        billingIssuesDetectedAt: Date? = nil,
+        ownershipType: PurchaseOwnershipType = .defaultValue
+    ) {
+        self.periodType = periodType
+        self.purchaseDate = purchaseDate
+        self.originalPurchaseDate = originalPurchaseDate
+        self.expiresDate = expiresDate
+        self.store = store
+        self.isSandbox = isSandbox
+        self.unsubscribeDetectedAt = unsubscribeDetectedAt
+        self.billingIssuesDetectedAt = billingIssuesDetectedAt
+        self.ownershipType = ownershipType
+    }
+
+    var asTransaction: CustomerInfoResponse.Transaction {
+        return .init(purchaseDate: self.purchaseDate,
+                     originalPurchaseDate: self.originalPurchaseDate,
+                     transactionIdentifier: nil,
+                     store: self.store,
+                     isSandbox: self.isSandbox)
+    }
+
+}
+
+extension CustomerInfoResponse.Subscriber {
+
+    var allTransactionsByProductId: [String: CustomerInfoResponse.Transaction] {
```

Comment:
`CustomerInfo` was combining the dictionaries inside of "subscriptions" and "non_subscriptions", despite them having different formats. This is now made explicit through type conversions between the two.